src/formattedText.py

```diff
@@ -0,0 +1,127 @@
+# Copyright (c) 2015-present, Facebook, Inc.
+# All rights reserved.
+#
+# This source code is licensed under the BSD-style license found in the
+# LICENSE file in the root directory of this source tree. An additional grant
+# of patent rights can be found in the PATENTS file in the same directory.
+#
+# @nolint
+
+import re
+import curses
+
+class FormattedText(object):
+    """A piece of ANSI escape formatted text which responds
+    to str() returning the plain text and knows how to print
+    itself out using ncurses"""
+
+    ANSI_ESCAPE_FORMATTING = r'\x1b\[([^mK]*)[mK]'
+    BOLD_ATTRIBUTE = 1
+    UNDERLINE_ATTRIBUTE = 4
+
+    def __init__(self, text):
+        self.text = text
+
+        self.segments = re.split(self.ANSI_ESCAPE_FORMATTING, self.text)
+        #re.split will insert a empty string if there is a match at the beginning
+        #or it will return [string] if there is no match
+        #create the invariant that every segment has a formatting segment, e.g
+        #we will always have FORMAT, TEXT, FORMAT, TEXT
+        self.segments.insert(0, '')
+        self.plainText = ''.join(self.segments[1::2])
+
+
+    def __str__(self):
+        return self.plainText
+
+    @classmethod
+    def parseFormatting(cls, formatting):
+        """Parse ANSI formatting; the formatting passed in should be
+        stripped of the control characters and ending character"""
+        fore = -1 #-1 default means "use default", not "use white/black"
+        back = -1
+        other = 0
+        intValues = [int(value) for value in formatting.split(';') if value]
+        for code in intValues:
+            if code >= 30 and code <= 39:
```

Comment:
should these `30` and `40`'s be constants in the class? then we could at least explain what they mean and there would be less magic numbers here